abis_mapping/templates/survey_site_data_v2/mapping.py

```diff
@@ -101,13 +102,51 @@ def apply_validation(
                     plugins.mutual_inclusion.MutuallyInclusive(
                         field_names=["relatedSiteID", "relationshipToRelatedSite"],
                     ),
+                    # When relationshipToRelatedSite is 'partOf'
+                    # then the relatedSiteID must exist as a siteID in the same template.
+                    plugins.row_constraint.RowConstraintSideInput(
+                        side_inputs={"allSiteIDs": self.extract_site_ids(data).keys()},
+                        original_schema=frictionless.Schema.from_descriptor(self.schema()),
+                        formula=(
+                            "relatedSiteID in allSiteIDs "
+                            "if relatedSiteID and relationshipToRelatedSite.lower().replace(' ', '')=='partof' "
+                            "else True"
+                        ),
+                    ),
                 ],
             )
         )
 
         # Return validation report
         return report
 
+    def extract_site_ids(
+        self,
+        data: base.types.ReadableType,
+    ) -> dict[str, Literal[True]]:
```

Comment:
no need to change this I'm just curious, is there a reason we don't return `set[str]` here? is it to be consistent with the other extract methods? then my my question is why don't those return `set[str]` instead of `dict[str, True]`?

Reply:
consistency. Idea was to return map types from extractor methods. Specifically chose `dict[str, True]` here, since it is more specific than `dict[str, bool]` while being compatible by any consumer that expects a boolean.